com.unity.netcode.gameobjects/Editor/CodeGen/NetworkBehaviourILPP.cs

```diff
@@ -1889,6 +1946,109 @@ private void InjectWriteAndCallBlocks(MethodDefinition methodDefinition, CustomA
             instructions.ForEach(instruction => processor.Body.Instructions.Insert(0, instruction));
         }
 
+        private void GenerateVariableInitialization(TypeDefinition type)
+        {
+            var method = new MethodDefinition(
+                k_NetworkBehaviour___initializeVariables,
+                MethodAttributes.Family | MethodAttributes.Virtual | MethodAttributes.HideBySig,
+                m_MainModule.TypeSystem.Void);
+
+            var processor = method.Body.GetILProcessor();
+
+            method.Body.Variables.Add(new VariableDefinition(m_MainModule.TypeSystem.Boolean));
+
+            processor.Emit(OpCodes.Nop);
+
+            foreach (var fieldDefinition in type.Fields)
+            {
+                FieldReference field = fieldDefinition;
+                if (type.HasGenericParameters)
+                {
+                    var genericType = new GenericInstanceType(fieldDefinition.DeclaringType);
+                    foreach (var parameter in fieldDefinition.DeclaringType.GenericParameters)
+                    {
+                        genericType.GenericArguments.Add(parameter);
+                    }
+                    field = new FieldReference(fieldDefinition.Name, fieldDefinition.FieldType, genericType);
+                }
+                if (field.FieldType.IsSubclassOf(m_NetworkVariableBase_TypeRef))
+                {
+                    processor.Emit(OpCodes.Ldarg_0);
```

Comment:
I usually find inline pseudo-C# code helpful understanding what was tried to be achieved with IL emits.

Reply:
Done.

Reply:
Nice... now I can understand what the OpCodes actually mean! 😸 
(TILT)